hoodie-common/src/main/java/com/uber/hoodie/common/util/SpillableMapUtils.java

```diff
@@ -0,0 +1,155 @@
+   */
+  private static DiskBasedMap.FileEntry readInternal(RandomAccessFile file, long valuePosition, int valueLength) throws IOException {
+    file.seek(valuePosition);
+    long crc = file.readLong();
+    long timestamp = file.readLong();
+    int keySize = file.readInt();
+    int valueSize = file.readInt();
+    byte [] key = new byte[keySize];
+    file.read(key, 0, keySize);
+    byte [] value = new byte[valueSize];
+    if(!(valueSize == valueLength)) {
+      throw new HoodieCorruptedDataException("unequal size of payload written to external file");
+    }
+    file.read(value, 0, valueSize);
+    long crcOfReadValue = SpillableMapUtils.generateChecksum(value);
+    if(!(crc == crcOfReadValue)) {
+      throw new HoodieCorruptedDataException("checksum of payload written to external disk does not match");
+    }
+    return new DiskBasedMap.FileEntry(crc, keySize, valueSize, key, value, timestamp);
+  }
+
+  /**
+   * Write Value and other metadata necessary to disk. Each entry looks as follows:
+   *
+   * |crc|timestamp|sizeOfKey|SizeOfValue|key|value|
+   *
+   * @param outputStream
+   * @param fileEntry
+   * @return
+   * @throws IOException
+   */
+  public static long write(SizeAwareDataOutputStream outputStream, DiskBasedMap.FileEntry fileEntry) throws IOException {
+    return writeInternal(outputStream, fileEntry);
+  }
+
+
+  private static long writeInternal(SizeAwareDataOutputStream outputStream, DiskBasedMap.FileEntry fileEntry)
+      throws IOException {
+    outputStream.writeLong(fileEntry.getCrc());
+    outputStream.writeLong(fileEntry.getTimestamp());
+    outputStream.writeInt(fileEntry.getSizeOfKey());
+    outputStream.writeInt(fileEntry.getSizeOfValue());
+    outputStream.write(fileEntry.getKey());
+    outputStream.write(fileEntry.getValue());
+    outputStream.flush();
+    return outputStream.getSize();
+  }
+
+  /**
+   * Generate a checksum for a given set of bytes
+   * @param data
+   * @return
+   */
+  public static long generateChecksum(byte [] data) {
+    CRC32 crc = new CRC32();
+    crc.update(data);
+    return crc.getValue();
+  }
+
+  /**
+   * Estimate the value of the payload by serializing the contents
+   * @param value
+   * @param schema
+   * @param <R>
+   * @return
+   * @throws IOException
+   */
+  public static <R> int estimateValueSize(R value, Schema schema) throws IOException {
+    HoodieRecord payload = (HoodieRecord) value;
+    byte [] val = HoodieAvroUtils.avroToBytes((GenericRecord) payload.getData().getInsertValue(schema).get());
+    return val.length;
+  }
+
+  /**
+   * Utility method to convert bytes to HoodieRecord using schema and payload class
+   * TODO : Make this utility generic enough to be used in HoodieRecordScanner
+   * @param bytes
+   * @param schema
+   * @param payloadClazz
+   * @param <R>
+   * @return
+   * @throws IOException
+   */
+  private static <R> R convertBytesToRecord(byte [] bytes, Schema schema, String payloadClazz) throws IOException {
```

Comment:
fmt: `byte[] bytes` ?

Reply:
Seems like this method could be shared with some code in CompactedRecordScanner as well?

Reply:
Yeah, added a TODO, will look into this.

Reply:
done